bin/pav.py

```diff
@@ -93,55 +96,63 @@ def record_factory(*fargs, **kwargs):
+                                           maxBytes=1024 ** 2,
+                                           backupCount=3)
+        file_handler.setFormatter(logging.Formatter(pav_cfg.log_format,
+                                                    style='{'))
+        file_handler.setLevel(getattr(logging,
+                                      pav_cfg.log_level.upper()))
+        root_logger.addHandler(file_handler)
 
     # The root logger should pass all messages, even if the handlers
     # filter them.
     root_logger.setLevel(logging.DEBUG)
 
     # Setup the result logger.
     # Results will be logged to both the main log and the result log.
-    result_logger = logging.getLogger('results')
-    result_handler = RotatingFileHandler(filename=str(pav_cfg.result_log),
-                                         # 20 MB
-                                         maxBytes=20 * 1024 ** 2,
-                                         backupCount=3)
-    result_handler.setFormatter(logging.Formatter("{asctime} {message}",
-                                                  style='{'))
-    result_logger.setLevel(logging.INFO)
-    result_logger.addHandler(result_handler)
+    try:
+        pav_cfg.result_log.touch()
+    except (PermissionError, FileNotFoundError) as err:
+        output.dbg_print("Could write create result log at '{}': {}"
+                         .format(pav_cfg.result_log, err))
+    else:
+        result_logger = logging.getLogger('results')
+        result_handler = RotatingFileHandler(filename=str(pav_cfg.result_log),
+                                             # 20 MB
+                                             maxBytes=20 * 1024 ** 2,
+                                             backupCount=3)
+        result_handler.setFormatter(logging.Formatter("{asctime} {message}",
+                                                      style='{'))
+        result_logger.setLevel(logging.INFO)
+        result_logger.addHandler(result_handler)
 
     # Setup the exception logger.
     # Exceptions will be logged to this directory, along with other useful info.
     exc_logger = logging.getLogger('exceptions')
     try:
-        if not pav_cfg.exception_log.exists():
-            pav_cfg.exception_log.parent.mkdir(
-                mode=0o775,
-                parents=True,
-                #exist_ok=True  # Doesn't work in python 3.4 (added in 3.5)
-            )
-    except (PermissionError, OSError, IOError, FileExistsError) as err:
-        pavilion.output.dbg_print("Could not create exception log")
-
-    exc_handler = RotatingFileHandler(
-        filename=pav_cfg.exception_log.as_posix(),
-        maxBytes=20 * 1024 **2,
-        backupCount=3,
-    )
-    exc_handler.setFormatter(logging.Formatter(
-        "{asctime} {message}",
-        style='{',
-    ))
-    exc_logger.setLevel(logging.ERROR)
-    exc_logger.addHandler(exc_handler)
+        pav_cfg.exception_log.touch()
```

Comment:
Still getting this error when running the Travis tests:

```
- Could not create exception log at '~/.pavilion/working_dir/exceptions.log': [Errno 2] No such
- file or directory: '~/.pavilion/working_dir/exceptions.log'
```

Reply:
oh, hah.